weed/storage/needle_map_memory.go

```diff
@@ -79,3 +83,49 @@ func (nm *NeedleMap) Destroy() error {
 	nm.Close()
 	return os.Remove(nm.indexFile.Name())
 }
+
+func (nm *NeedleMap) UpdateNeedleMap(v *Volume, indexFile *os.File, opts *opt.Options) error {
+	if v.nm != nil {
+		v.nm.Close()
+		v.nm = nil
+	}
+	defer func() {
+		if v.tmpNm != nil {
+			v.tmpNm.Close()
+			v.tmpNm = nil
+		}
+	}()
+	nm.indexFile = indexFile
+	stat, err := indexFile.Stat()
+	if err != nil {
+		glog.Fatalf("stat file %s: %v", indexFile.Name(), err)
+		return err
+	}
+	nm.indexFileOffset = stat.Size()
+	v.nm = nm
```

Comment:
Between line 90 and line 105, `v.nm` is nil. If request comes in then, there would be nil pointer exception.

Same for `weed/storage/needle_map_leveldb.go` UpdateNeedleMap implementation.

Reply:
When loading volume, no request will be handled.
https://github.com/seaweedfs/seaweedfs/blob/10414fd81c2744b2b031f4d2f8a935b614d702f1/weed/storage/volume_vacuum.go#L113-L116

Reply:
When loading a new volume, no request for this volume will be handled too.